fix(sensor): change configuration and delta in adxl - PerformSelftest

PandiasChild opened the pull request.

PandiasChild committed:
fix(sensor): change configuration and delta in adxl - PerformSelftest

DavidFederl commented on src/sensor/adxl345b/Adxl345b.c:
```diff
@@ -352,30 +352,40 @@ adxl345bErrorCode_t adxl345bPerformSelfTest(adxl345bSensorConfiguration_t sensor
-                                                   ADXL345B_BW_RATE_25);
+
+    /* standard mode, 100Hz */
+    errorCode =
+        adxl345bWriteConfigurationToSensor(sensor, ADXL345B_REGISTER_BW_RATE, ADXL345B_BW_RATE_100);
     if (errorCode != ADXL345B_NO_ERROR) {
         return errorCode;
     }
-    /* right adjusted, 16G range */
-    errorCode = adxl345bChangeMeasurementRange(sensor, ADXL345B_16G_RANGE);
+
+    /* disable selftest_bit, enable 16g full resolution  */
+    errorCode =
+        adxl345bWriteConfigurationToSensor(sensor, ADXL345B_REGISTER_DATA_FORMAT, 0b00001011);
```
use original adxl345bChangeMeasurementRange `adxl345bChangeMeasurementRange(...)` method and add function to `setSelfTest(...)` to enable force.

DavidFederl commented on src/sensor/adxl345b/Adxl345b.c:
```diff
@@ -419,28 +429,28 @@ adxl345bErrorCode_t adxl345bPerformSelfTest(adxl345bSensorConfiguration_t sensor
+    int avgSampleWithoutForceZ = sumSamplesWithoutForceZ / numberOfSamples;
 
-    /* enable self test force */
+    /* enable selftest_bit, while keeping 16g full resolution  */
```
```suggestion
    /* set selftest_bit to enable force, while keeping 16g full resolution  */
```

PandiasChild committed:
refactor(sensor): add internal function to set seltest-bit to enable force

PandiasChild committed:
feat(sensor): add function to activate Measurement Mode

PandiasChild committed:
refactor(sensor): make SetSelftestBit two functions enable- and disableSelftestForce"

PandiasChild committed:
perf(sensor): change sampling order and ammount

PandiasChild committed:
fix(sensor): fix CalculateCalibrationOffset

BREAKING CHANGE:
- adxl345bConvertDataXYZ now requires sensor as parameter
- adxl345bPerformSelfTest returns average instead of delta parameter[out]

PandiasChild committed:
feat(sensor): add functions to setResolutionmodes

- add adxl345bInternalSetRegisterBits and
 adxl345bInternalClearRegisterBits and Bitmasks for better readability
 - differenciate between 3V3 and 2V5 scalefactors in ranges as well as tenBit and fullRes

PandiasChild committed:
fix(sensor): fix adxl345bInternalConvertLSBtoGValue

 - add Sensorconfiguration to parameter
 - differenciate between tenBit and fullRes scale Factor

DavidFederl commented on src/sensor/adxl345b/Adxl345b.c:
```diff
@@ -244,328 +258,286 @@ adxl345bErrorCode_t adxl345bGetMultipleMeasurements(adxl345bSensorConfiguration_
         }
 
-        if (interrupt_source & 0b10000000) {
+        if ((interrupt_source & ADXL345B_BITMASK_DATA_READY) == ADXL345B_BITMASK_DATA_READY) {
```
Should be sufficient:
```suggestion
        if (interrupt_source & ADXL345B_BITMASK_DATA_READY) {
```

DavidFederl commented on src/sensor/adxl345b/Adxl345b.c:
```diff
@@ -8,13 +8,26 @@
 #include "Time.h"
 #include "include/Adxl345bTypedefs.h"
 
+#include "math.h"
+
```
If you start splitting the headers, I would suggest to move the C standard header to the top.

DavidFederl reviewed (approved):
Addressed changes are optional, feel free to apply them, or not.

PandiasChild commented:
@DavidFederl  @glencoe can we please merge this into main?
